src/runtime/FinalizationRegistryObject.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef __EscargotFinalizationRegistryObject__
+#define __EscargotFinalizationRegistryObject__
+
+#include "runtime/Object.h"
+
+namespace Escargot {
+
+class FinalizationRegistryObject : public Object {
+public:
+    struct FinalizationRegistryObjectItem : public gc {
+        EncodedValue weakRefTarget;
+        EncodedValue heldValue;
+        EncodedValue unregisterToken;
+
+        void* operator new(size_t size);
+        void* operator new[](size_t size) = delete;
```

Comment:
You don't have to overload `new` operators because gc will track all the internal element in default.
Instead, I think that its better to add a constructor function for `FinalizationRegistryObjectItem` for initialization of each member.